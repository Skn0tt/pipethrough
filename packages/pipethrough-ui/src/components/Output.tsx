import * as React from "react";
import Paper from "@material-ui/core/Paper";
import styled from "styled-components";
import * as R from "ramda";

const Container = styled(Paper)`
  padding: 1px 10px 0px;
  height: 120px;
`;

const Header = styled.h1`
  margin-left: 5px;
`;

const BlackPaper = styled(Paper)`
  color: white;
  overflow-wrap: break-word;
  padding: 10px;
`;

type OwnProps = {
  title: string;
  text: string;
};

const Output: React.SFC<OwnProps> = props => {
  const { title, text } = props;

  return (
    <Container>
      <Header>{title}</Header>
      <BlackPaper style={{ backgroundColor: "#424242" }}>
        {text.split("\n").map((t, i) => (
          <React.Fragment key={i}>
            {t}
            <br />
          </React.Fragment>
        ))}
      </BlackPaper>
    </Container>
  );
};

export default Output;
