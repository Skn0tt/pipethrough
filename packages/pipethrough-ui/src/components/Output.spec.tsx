import * as React from "react";
import { shallow, mount } from "enzyme";
import Output from "./Output";

describe("Output", () => {
  const props = {
    text: "Hallo",
    title: "Title"
  };
  const comp = mount(<Output {...props} />);

  it("renders as expected", () => {
    expect(comp.render()).toMatchSnapshot();
  });

  it("show title and text", () => {
    const text = comp.text();
    expect(text).toContain(props.text);
    expect(text).toContain(props.title);
  });
});
