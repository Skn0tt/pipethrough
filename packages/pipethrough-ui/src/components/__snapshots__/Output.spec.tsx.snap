// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output renders as expected 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa dyfwjK",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-bwzfXH drzoWa",
        },
        "children": Array [
          Object {
            "data": "Title",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h1",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-htpNat ialmjg",
            "style": "background-color: rgb(66, 66, 66);",
          },
          "children": Array [
            Object {
              "data": "Hallo",
              "next": Object {
                "attribs": Object {},
                "children": Array [],
                "name": "br",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
            Object {
              "attribs": Object {},
              "children": Array [],
              "name": "br",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "data": "Hallo",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-htpNat ialmjg",
          "style": "background-color: rgb(66, 66, 66);",
        },
        "children": Array [
          Object {
            "data": "Hallo",
            "next": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "br",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
          Object {
            "attribs": Object {},
            "children": Array [],
            "name": "br",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "data": "Hallo",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-bwzfXH drzoWa",
          },
          "children": Array [
            Object {
              "data": "Title",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
